aiida/cmdline/commands/cmd_verdi.py

```diff
@@ -14,36 +14,27 @@
 
 import click
 
-from aiida.cmdline.params import options
+from aiida.cmdline.params import options, types
 from aiida.common.extendeddicts import AttributeDict
-from aiida.common import exceptions
 
 
 @click.group(context_settings={'help_option_names': ['-h', '--help']})
-@options.PROFILE()
+@options.PROFILE(type=types.ProfileParamType(load_profile=True))
 @click.version_option(None, '-v', '--version', message="AiiDA version %(version)s")
 @click.pass_context
 def verdi(ctx, profile):
     """The command line interface of AiiDA."""
-    from aiida.manage.configuration import get_config, load_profile
+    from aiida.manage.configuration import get_config, get_profile
 
     if ctx.obj is None:
         ctx.obj = AttributeDict()
 
     config = get_config(create=True)
```

Comment:
just wondering - the profile option will already have loaded the config.
Should we be passing the config in the context instead?

Reply:
This would require us to create the `ctx.obj` in the option type as well, binding it tightly to this concept. On top of that, we still have to keep it here, because if the `--profile` is not explicitly used, since there is no default, the `ProfileParamType.convert` won't even be called. So I prefer to just load the config here. If it has already been called before, this will just return the same instance, since it is a global and the call will be very cheap